### PR TITLE
Misc

### PR DESCRIPTION
**Pull request by jschueller**
- rework LinearModelStepwiseAlgorithm api
- #1772 
- #1776 
- #1784 DistributionFactory: Reject constant/nan samples
- #1787
- #1790
- #1793
- #1794
- #1795
- Sample: Drop FromCSV->FromText fallback
- started generic Distribution test (https://openturns.discourse.group/t/automated-consistency-tests-of-a-distribution/82/7)

### CODE REVIEW

**Comment by jschueller** [comment]
maybe it's time

**Review by sofianehaddad — APPROVED** [review]

**Comment by sofianehaddad** [comment]
@jschueller how could we check `locally` that #1794 is fixed?

**Comment by jschueller** [comment]
I guess rebuild the doc

**Comment by sofianehaddad** [comment]
Hmm I have also troubles with #1795 
History convergences is "constant" and axis incoherent 
image

I don't think this is what we are expecting....

File: lib/src/Base/Geom/IntervalMesher.cxx [hunk 1]
```diff
@@ -192,6 +192,10 @@ Mesh IntervalMesher::build(const Interval & interval,
   } // dimension == 2
   if (dimension == 3)
   {
+    // FIXME: https://github.com/openturns/openturns/issues/1670
+    if (diamond)
+      throw NotYetImplementedException(HERE) << "In IntervalMesher::build with 3-d/diamond=true";
```
Comment (regislebrun) [inline lib/src/Base/Geom/IntervalMesher.cxx:197]: I started a branch to fix this one, but it is not so easy...

File: lib/src/Base/Stat/CovarianceMatrix.cxx [hunk 2]
```diff
@@ -45,6 +45,13 @@ CovarianceMatrix::CovarianceMatrix(const MatrixImplementation & i)
   // Nothing to do
 }
 
+/* Constructor from SymmetricMatrix */
+CovarianceMatrix::CovarianceMatrix(const SymmetricMatrix & sym)
```
Comment (regislebrun) [inline lib/src/Base/Stat/CovarianceMatrix.cxx:49]: Do we really need that? Isn't it supposed to be done by inheritance?

File: lib/src/Base/Stat/CorrelationMatrix.cxx [hunk 3]
```diff
@@ -46,6 +46,13 @@ CorrelationMatrix::CorrelationMatrix(const MatrixImplementation & i)
   // Nothing to do
 }
 
+/* Constructor from SymmetricMatrix */
+CorrelationMatrix::CorrelationMatrix(const SymmetricMatrix & sym)
```
Comment (regislebrun) [inline lib/src/Base/Stat/CorrelationMatrix.cxx:50]: Do we really need that? Isn't it supposed to be done by inheritance?

File: lib/src/Base/Stat/SampleImplementation.cxx [hunk 4]
```diff
@@ -313,9 +313,8 @@ CLASSNAMEINIT(SampleImplementation)
 SampleImplementation SampleImplementation::BuildFromCSVFile(const FileName & fileName,
     const String & csvSeparator)
 {
-  if (csvSeparator == " ") throw InvalidArgumentException(HERE) << "Error: the space separator is not compatible for CSV file.";
-
-
+  if (csvSeparator == " ")
```
Comment (regislebrun) [inline lib/src/Base/Stat/SampleImplementation.cxx:316]: Maybe we should also add tab in the list of forbidden separators?

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelStepwiseAlgorithm.cxx [hunk 5]
```diff
@@ -155,21 +128,25 @@ LinearModelStepwiseAlgorithm::Direction LinearModelStepwiseAlgorithm::getDirecti
 }
 
 /* Penalty accessors */
+void LinearModelStepwiseAlgorithm::setPenalty(const Scalar penalty)
+{
+  penalty_ = penalty;
```
Comment (regislebrun) [inline lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelStepwiseAlgorithm.cxx:133]: Any value is allowed for penalty?

File: python/test/t_Distribution_std.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    CDF_up = distribution.computeCDF(interval.getUpperBound())
+    CDF_low = distribution.computeCDF(interval.getLowerBound())
+    computed_probability = CDF_up - CDF_low
+    if distribution.getName() != "Histogram":
+        ott.assert_almost_equal(probability, computed_probability, 1e-5, 1e-5, str(distribution))
+    
+    # MinimumVolumeLevelSet
+    probability = 0.9
+    levelSet, threshold = distribution.computeMinimumVolumeLevelSetWithThreshold(probability)
+    x = distribution.computeQuantile(1.0 - (1.0 - probability) / 2.0)
```
Comment (regislebrun) [inline python/test/t_Distribution_std.py:57]: IT IS WRONG! It assumes that the minimum volume level set is a segment symmetrical wrt the median.I already fixed it and sent the fix to @mbaudin47. I don't remember on which channel, probably as a comment on the repo where you picked the code.

**Review by regislebrun — CHANGES_REQUESTED** [review]
Only one serious issue, otherwise only surface comments.

File: python/test/t_Distribution_std.py (hunk 6 above)
Comment (sofianehaddad) [reply python/test/t_Distribution_std.py:57]: image

File: lib/src/Base/Stat/CovarianceMatrix.cxx (hunk 2 above)
Comment (jschueller) [reply lib/src/Base/Stat/CovarianceMatrix.cxx:49]: Yes, I tried a the c++ level

File: lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelStepwiseAlgorithm.cxx (hunk 5 above)
Comment (jschueller) [reply lib/src/Uncertainty/Algorithm/MetaModel/LinearModel/LinearModelStepwiseAlgorithm.cxx:133]: I think I should be positive

**Comment by jschueller** [comment]
> Hmm I have also troubles with #1795
> History convergences is "constant" and axis incoherent
> image
> 
> I don't think this is what we are expecting....

I fixed that too

**Comment by sofianehaddad** [comment]
For #1795 we can improve last graph in `ego` and illustrate the usage of `kwargs` in `View`
```
view = viewer.View(graph, axes_kw={"xticks": range(0, result.getIterationNumber(), 5)})
plt.show()
```
This avoids having non integer values for iteration number as illustrated below:
image

**Comment by jschueller** [comment]
ok, applied

File: lib/src/Base/Geom/IntervalMesher.cxx (hunk 1 above)
Comment (jschueller) [reply lib/src/Base/Geom/IntervalMesher.cxx:197]: I dont think its urgent

**Pull request merged**